### PR TITLE
First stage of navigation

### LINKED ISSUE #33: 4. Point to point navigation
Implement the basic logic of the point to point service.

- Research path planning algorithms.
- Perform some experiments to determine suitability, report results here.
- Integrate into navigation state machine.
- Create a test for requesting the navigation service.

### PR DESCRIPTION
**Pull request by LiamWellacott**
- Allows the controller to request movement of the robot to a given goal pose. Closes #33 
- map of the simulated environment
- launch file for navigation

### CODE REVIEW

File: navigation/scripts/navigation.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+
+import rospy
+
+IDLE = 0
```
Comment (NicolayP) [inline navigation/scripts/navigation.py]: Always dangerous to play with global variables, better have a class that contains all the information imo.

File: navigation/scripts/nav.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+from actionlib import SimpleActionClient
+
+from navigation.srv import NavigateTo, NavigateToResponse
+
+from move_base_msgs.msg import MoveBaseGoal, MoveBaseAction
+from actionlib_msgs.msg import GoalStatusArray
+from geometry_msgs.msg import PoseWithCovarianceStamped, Pose
+
+IDLE = 0
```
Comment (NicolayP) [inline navigation/scripts/nav.py]: Use a enum or put the navigation state inside the class

File: navigation/scripts/nav.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+        # initialise state variables
+        self._reset()
+
+        # services
+        self.robot_pose = Pose()
+        s = rospy.Service('navigate_to', NavigateTo, self.setNavigate)
+
+        # sub
+        rospy.Subscriber('robot_pose', PoseWithCovarianceStamped, self._pose_callback)
```
Comment (NicolayP) [inline navigation/scripts/nav.py:38]: this will be fed by SLAM ?

File: navigation/scripts/nav.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    ### service functions
+    def setExplore(self):
+        self.state = EXPLORING
+
+    def setNavigate(self, msg):
+        self._reset()
+        self.state = NAVIGATING
+        self.goal_pose = msg.pose
+        return NavigateToResponse(True)
```
Comment (NicolayP) [inline navigation/scripts/nav.py:66]: the service is never going to fail ?

File: navigation/scripts/nav.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        elif self.state == NAVIGATING:
+            self.navigate()
+        elif self.state == FOLLOWING:
+            self.follow()
+        elif self.state == GUIDING:
+            self.guide()
+
+def main():
+
+    rospy.init_node('navigation', anonymous=True)
```
Comment (NicolayP) [inline navigation/scripts/nav.py:139]: you can create the node inside the class, makes the instantiation cleaner

File: navigation/scripts/nav.py [hunk 6]
```diff
@@ -28,7 +29,9 @@ def __init__(self):
         self.robot_pose = Pose()
         s = rospy.Service('navigate_to', NavigateTo, self.setNavigate)
 
-        # sub
+        # pub/sub
+        self.nav_done_pub = rospy.Publisher('nav_done', Bool, queue_size=10)
```
Comment (NicolayP) [inline navigation/scripts/nav.py:36]: change the nav service to just a publisher no? 
Regards,
Camenbert fromage

File: navigation/scripts/nav.py (hunk 3 above)
Comment (LiamWellacott) [reply navigation/scripts/nav.py:38]: yes

File: navigation/scripts/nav.py (hunk 4 above)
Comment (LiamWellacott) [reply navigation/scripts/nav.py:66]: can change to empty response, or can leave it for future if we want the nav to be able to reject requests

File: navigation/scripts/nav.py (hunk 5 above)
Comment (LiamWellacott) [reply navigation/scripts/nav.py:139]: intentionally outside to allow reusing the class without initialising a node

File: navigation/scripts/nav.py (hunk 6 above)
Comment (LiamWellacott) [reply navigation/scripts/nav.py:36]: this is a topic for notifying the rest of the system that the navigation has completed, I think it makes sense as is?

**Pull request merged**